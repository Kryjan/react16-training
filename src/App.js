import React, { Component } from 'react';
import { BookOverview } from './components/BookOverview';
import './App.css';
const books = [
  { id: 1, author: 'David Flanagan', title: 'JS: Definitive Guide' },
  { id: 2, author: 'Douglas Crockford', title: 'JS: The Good Parts' },
]

export class App extends Component {


  render() {
    return (
      <div className="book-app">
        <BookOverview books={books}></BookOverview>
      </div>
    );
  }
}