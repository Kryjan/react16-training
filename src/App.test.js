import React from "react";
import ReactDOM from "react-dom";
import App from "./App";

import { shallow, mount, render } from "enzyme";

describe("renders without crashing", () => {
  it("when default approach used", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("when enzyme", () => {
    it("shallow used", () => {
      shallow(<App />);
    });

    it("mount used", () => {
      mount(<App />);
    });
  });

  describe("when enzyme snapshots taken with", () => {
    it("shallow", () => {
      expect(shallow(<App />)).toMatchSnapshot();
    });

    it("mount", () => {
      expect(mount(<App />)).toMatchSnapshot();
    });

    it("render", () => {
      expect(render(<App />)).toMatchSnapshot();
    });
  });
});

describe("renders welcome message", () => {
  const wrapper = shallow(<App />);
  const welcome = <h1 className="App-title">Welcome to React</h1>;

  it('checked with plain enzyme', () => {
    expect(wrapper.contains(welcome)).toEqual(true);
  });

  it('checked with jest-enzyme', () => {
    expect(wrapper).toContainReact(welcome);
  });
});
