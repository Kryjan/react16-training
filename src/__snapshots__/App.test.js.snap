// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when enzyme snapshots taken with mount 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
  </div>
</App>
`;

exports[`renders without crashing when enzyme snapshots taken with render 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    >
      
    </img>
    <h1
      class="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    class="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`renders without crashing when enzyme snapshots taken with shallow 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
